### ISSUE TITLE
JSONParser hates the word "clear"?

### ISSUE THREAD

**Issue by deldrid1** [electricimp/JSONParser#9]
```squirrel
#require "JSONParser.class.nut:1.0.0"

local json = @"{
    ""clear"": 157
}"

local tbl = {
    "clear": 157
}

// This works fine
server.log(http.jsondecode(json))
server.log(http.jsondecode(http.jsonencode(tbl)))


// Both of these fail
server.log(JSONParser.parse(json))
server.log(JSONParser.parse(http.jsonencode(tbl)))
```

```
2018-09-13 18:51:32 -04:00 | [Status] | Agent started.
2018-09-13 18:51:32 -04:00 | [Agent] | (table : 0x7efb4b3658f0)
2018-09-13 18:51:32 -04:00 | [Agent] | (table : 0x7efb194c9380)
2018-09-13 18:51:32 -04:00 | [Agent] | ERROR: JSON Syntax Error near `: 157 }`
2018-09-13 18:51:32 -04:00 | [Agent] | ERROR:   in parse ...imp#jsonparser.class.nut#1.0.0:286
2018-09-13 18:51:32 -04:00 | [Agent] | ERROR:   from main agent_code:18
```

Any ideas?  I've noticed that it doesn't really seem to matter where in the table `clear` shows up... (i.e. it can be nested, etc.).  Reviewing the code though I don't see any particular reason this would have an issue...

**Comment by ElectricImpSampleCode** [comment]
I've come to the party very late, but I've figured this one out. When checking for duplicate key names in the decoded JSON, the library uses the construct `if (key in container)...` where _container_ is the table that'll be returned by the parser. Unfortunately, this construct evaluates to `true` if _key_ is assigned a name that matches one the table methods, eg. clear (as you found, @deldrid1), rawin, rawset, etc.
I'm updating the library to fix this (Squirrel bug?) but I thought you might like to know the source of the problem.

Tony

**Comment by smittytone** [comment]
Fix now merged into version 1.0.1. Marking this issue closed.
